### PR TITLE
feat: add event_metadata parameter to EventBusProducer send API

### PR DESCRIPTION
**Pull request by rgraber**
**Description:**
Update the send() API to optionally take an EventsMetadata object (which are sent by OpenEdxPublicSignals whenever they send an event). This object contains the information necessary to add the correct CloudEvent headers to an event.

This can be merged in any order with https://github.com/openedx/event-bus-kafka/pull/87

**ISSUE:** https://github.com/openedx/openedx-events/issues/77

**Reviewers:**
- [ ] tag reviewer
- [ ] tag reviewer

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant
solutions, hacks, quick-and-dirty implementations, concerns about
migrations, etc.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -14,6 +14,12 @@ Change Log
 Unreleased
 ----------
 
+[3.2.0] - 2022-11-30
+--------------------
+Changed
+~~~~~~~
+* Add event_metadata parameter to EventBusProducer send API
```
Comment (timmc-edx) [inline CHANGELOG.rst:21]:
```suggestion
* Add event_metadata parameter to EventBusProducer send API. **Note:** This parameter cannot be used until the caller also upgrades to a version of openedx-events that supports this parameter.
```
...and we probably want a matching comment in the openedx-events changelog about updating implementation libraries.

Comment (timmc-edx) [reply CHANGELOG.rst:21]: Sorry, just realized that this *is* openedx-events. Updating suggestion...

Comment (rgraber) [reply CHANGELOG.rst]: I adopted the idea of the suggestion but used somewhat different wording that I think is more accurate. LMK what you think.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -14,6 +14,12 @@ Change Log
 Unreleased
 ----------
 
+[3.2.0] - 2022-11-30
+--------------------
+Changed
+~~~~~~~
+* Add event_metadata parameter to EventBusProducer send API.  **Note:** Calling this method with the new argument will throw an exception unless the EventBusProducer implementation used also upgraded to support the argument.
```
Comment (robrap) [inline CHANGELOG.rst:21]: Nit: I think you are saying the following? If so, maybe this comment is more clear? You be the judge.
```suggestion
* Add event_metadata parameter to EventBusProducer send API.  **Note:** Calling this method with the new argument will throw an exception with older versions of this library that don't yet support the parameter.
```

File: CHANGELOG.rst (hunk 1 above)
Comment (timmc-edx) [reply CHANGELOG.rst]: Yeah, seems fine.

**Review by timmc-edx — APPROVED** [review]

File: CHANGELOG.rst (hunk 2 above)
Comment (robrap) [reply CHANGELOG.rst:21]: I reviewed the wrong PR. :)

**Review by robrap — APPROVED** [review]
Minor nit.

**Pull request merged**